src/Vehicle/VehicleBatteryFactGroup.cc

```diff
@@ -55,22 +64,76 @@ VehicleBatteryFactGroup::VehicleBatteryFactGroup(uint8_t batteryId, QObject* par
     _addFact(&_timeRemainingStrFact,        _timeRemainingStrFactName);
     _addFact(&_chargeStateFact,             _chargeStateFactName);
     _addFact(&_instantPowerFact,            _instantPowerFactName);
+    _addFact(&_timeUntilNextThresholdFact,  _timeUntilNextThresholdFactName);
+    _addFact(&_nextThresholdNameFact,       _nextThresholdNameFactName);
+    _addFact(&_avgCurrentLastMinuteFact,    _avgCurrentLastMinuteFactName);
 
-    _batteryIdFact.setRawValue          (batteryId);
-    _batteryFunctionFact.setRawValue    (MAV_BATTERY_FUNCTION_UNKNOWN);
-    _batteryTypeFact.setRawValue        (MAV_BATTERY_TYPE_UNKNOWN);
-    _voltageFact.setRawValue            (qQNaN());
-    _currentFact.setRawValue            (qQNaN());
-    _mahConsumedFact.setRawValue        (qQNaN());
-    _temperatureFact.setRawValue        (qQNaN());
-    _percentRemainingFact.setRawValue   (qQNaN());
-    _timeRemainingFact.setRawValue      (qQNaN());
-    _chargeStateFact.setRawValue        (MAV_BATTERY_CHARGE_STATE_UNDEFINED);
-    _instantPowerFact.setRawValue       (qQNaN());
+    _batteryIdFact.setRawValue             (batteryId);
+    _batteryFunctionFact.setRawValue       (MAV_BATTERY_FUNCTION_UNKNOWN);
+    _batteryTypeFact.setRawValue           (MAV_BATTERY_TYPE_UNKNOWN);
+    _voltageFact.setRawValue               (qQNaN());
+    _currentFact.setRawValue               (qQNaN());
+    _mahConsumedFact.setRawValue           (qQNaN());
+    _temperatureFact.setRawValue           (qQNaN());
+    _percentRemainingFact.setRawValue      (qQNaN());
+    _timeRemainingFact.setRawValue         (qQNaN());
+    _chargeStateFact.setRawValue           (MAV_BATTERY_CHARGE_STATE_UNDEFINED);
+    _instantPowerFact.setRawValue          (qQNaN());
+    _timeUntilNextThresholdFact.setRawValue(qQNaN());
+    _nextThresholdNameFact.setRawValue     ("");
+    _avgCurrentLastMinuteFact.setRawValue  (qQNaN());
+
+    auto aviantSettings = qgcApp()->toolbox()->settingsManager()->aviantSettings();
+
+    Fact* batCapacityFact = vehicle->parameterManager()->getParameter(-1, "BAT1_CAPACITY");
+    Fact* batCritFact = vehicle->parameterManager()->getParameter(-1, "BAT_CRIT_THR");
+    Fact* batEmergenFact = vehicle->parameterManager()->getParameter(-1, "BAT_EMERGEN_THR");
+
+    int batCapacity = batCapacityFact ? batCapacityFact->rawValue().toInt() : 0;
+    double critThreshold = batCritFact ? batCritFact->rawValue().toDouble() : 0.0;
+    double emergenThreshold = batEmergenFact ? batEmergenFact->rawValue().toDouble() : 0.0;
+
+    _thresholds = {
```

Comment:
Maybe comment about choices regarding mAh vs voltage vs SoC

Reply:
I did not understand what you meant here

Reply:
i means about the possible discrepancy between voltage, consumed mAh and estiamted SoC. but not important